components/notes/RichTextEditor.tsx

```
@@ -0,0 +1,253 @@
+  }
+
+  if (status === "error") {
+    return (
+      <div className="flex flex-col justify-center items-center h-40 text-destructive bg-destructive/10 rounded-md">
+        <AlertTriangle className="h-8 w-8 mb-2" />
+        <p>Failed to load or save note. Please refresh.</p>
+      </div>
+    );
+  }
+
+  const basePlugins = useMemo(() => [
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)